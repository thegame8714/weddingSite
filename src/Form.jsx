import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import messages from './translation/sentences'

class Form extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      name: undefined,
      code: undefined
    }

    this.handleInputNameChange = this.handleInputNameChange.bind(this)
    this.handleInputCodeChange = this.handleInputCodeChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleInputNameChange(e) {
    this.setState({
      name: e.target.value
    })
  }

  handleInputCodeChange(e) {
    this.setState({
      code: e.target.value
    })
  }

  handleSubmit(e) {
    const { name, code } = this.state
    e.preventDefault()
    this.props.onSubmitClick(name, code)
  }

  render() {
    return (
      <form className="main_content_form">
        <div className="main_content_form_item">
          <label htmlFor="fullName">
            <FormattedMessage {...messages.name} />
          </label>
          <input
            name="fullName"
            type="text"
            data-test-input-name
            onChange={this.handleInputNameChange}
          />
        </div>
        <div className="main_content_form_item">
          <label htmlFor="secretCode">
            <FormattedMessage {...messages.secret_code} />
          </label>
          <input
            name="secretCode"
            type="password"
            data-test-input-code
            onChange={this.handleInputCodeChange}
          />
        </div>
        <div>
          <button
            type="submit"
            data-test-submit-button
            onClick={this.handleSubmit}
            className="main_content_button"
          >
            <FormattedMessage {...messages.go_to_payment_page} />
          </button>
        </div>
      </form>
    )
  }
}

Form.propTypes = {
  onSubmitClick: PropTypes.func.isRequired
}

export default Form
