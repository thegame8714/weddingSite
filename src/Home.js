import React from 'react'
import { FormattedMessage } from 'react-intl'
import messages from './translation/sentences'
import BankAccount from './BankAccount'
import Form from './Form'
import axios from 'axios'

class Home extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      showBankAccount: false,
      nameNotFound: false
    }

    this.onSubmitClick = this.onSubmitClick.bind(this)
    this.apiCall = this.apiCall.bind(this)
  }

  apiCall(name, code) {
    return axios.request({
      method: 'post',
      url: 'http://api.giadazanotti.com/addGift',
      data: {
        name,
        code
      },
      responseType: 'json'
    })
  }

  onSubmitClick(name, code) {
    return this.apiCall(name, code)
      .then(() => {
        this.setState({
          showBankAccount: true,
          nameNotFound: false
        })
      })
      .catch(() => {
        this.setState({
          nameNotFound: true
        })
      })
  }

  render() {
    return (
      <div>
        <div className="main_content_title">
          <FormattedMessage {...messages.title} />
        </div>
        <div className="main_content_text">
          <p>
            <FormattedMessage {...messages.welcome_text} />
          </p>
          <p>
            <FormattedMessage {...messages.we_cant_wait} />
          </p>
          <p>
            <FormattedMessage {...messages.dreamy_honeymoon} />
          </p>
          <p>
            <FormattedMessage {...messages.contribute} />
          </p>
        </div>
        <div className="main_content_thank_you_message">
          <FormattedMessage {...messages.thank_you} />
        </div>
        {this.state.nameNotFound && <span>WROONG!!!</span>}
        {this.state.showBankAccount ? (
          <BankAccount />
        ) : (
          <Form onSubmitClick={this.onSubmitClick} />
        )}
      </div>
    )
  }
}

export default Home
