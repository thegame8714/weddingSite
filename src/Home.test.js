import renderer from 'react-test-renderer'
import React from 'react'
import Home from './Home'
import { IntlProvider } from 'react-intl'
import { mount } from 'enzyme'
import sinon from 'sinon'
import Form from './Form'
const messages = require('./translation/en.json')

describe('Home component', () => {
  it('should render the home component', () => {
    const tree = renderer
      .create(
        <IntlProvider locale="en" messages={messages}>
          <Home />
        </IntlProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render the Form component', () => {
    const homeWrapper = mount(
      <IntlProvider locale="en" messages={messages}>
        <Home />
      </IntlProvider>
    )

    expect(homeWrapper.find(Form).length).toBe(1)
  })

  it('should render the field for the name', () => {
    const appWrapper = mount(
      <IntlProvider locale="en" messages={messages}>
        <Home />
      </IntlProvider>
    )

    expect(appWrapper.find('[data-test-input-name]').length).toBe(1)
  })

  it('should render the field for the security code', () => {
    const appWrapper = mount(
      <IntlProvider locale="en" messages={messages}>
        <Home />
      </IntlProvider>
    )

    expect(appWrapper.find('[data-test-input-code]').length).toBe(1)
  })

  it('should call the sendToAPI function when user clicks on the submit button', () => {
    const sendToAPISpy = sinon.spy()
    const formWrapper = mount(
      <IntlProvider locale="en" messages={messages}>
        <Form onSubmitClick={sendToAPISpy} />
      </IntlProvider>
    )

    const inputNameField = formWrapper.find('[data-test-input-name]')
    inputNameField.simulate('change', { target: { value: 'Fabio Salimbeni' } })

    const inputCodeField = formWrapper.find('[data-test-input-code]')
    inputCodeField.simulate('change', { target: { value: '12345' } })

    formWrapper.find('[data-test-submit-button]').simulate('click')

    expect(sendToAPISpy.calledOnce).toBe(true)
    expect(sendToAPISpy.calledWith('Fabio Salimbeni', '12345')).toBeTruthy()
  })

  it('should render the bank account where to send the money if the code and name are right', async () => {
    const onSubmitClickStub = sinon
      .stub(Home.prototype, 'apiCall')
      .returns(new Promise((res, err) => res()))

    const appWrapper = mount(
      <IntlProvider locale="en" messages={messages}>
        <Home />
      </IntlProvider>
    )
    const inputNameField = appWrapper.find('[data-test-input-name]')
    inputNameField.simulate('change', { target: { value: 'Fabio Salimbeni' } })

    const inputCodeField = appWrapper.find('[data-test-input-code]')
    inputCodeField.simulate('change', { target: { value: '12345' } })

    appWrapper.find('[data-test-submit-button]').simulate('click')

    await new Promise(resolve => setImmediate(resolve()))

    appWrapper.update()

    expect(appWrapper.find('.bank-account').length).toBe(1)
    onSubmitClickStub.restore()
  })
})
