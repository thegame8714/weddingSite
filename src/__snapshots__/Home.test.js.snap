// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render the home component 1`] = `
<div>
  <div
    className="main_content_title"
  >
    <span>
      Fabio & Giada's wedding
    </span>
  </div>
  <div
    className="main_content_text"
  >
    <p>
      <span>
        Welcome into our wedding page, handcrafted for you by us.
      </span>
    </p>
    <p>
      <span>
        We can't wait to share this day with you!
      </span>
    </p>
    <p>
      <span>
        We haven't set up a proper giftlist but we are collecting money to live our dreamy honeymoon and live forever happily after!
      </span>
    </p>
    <p>
      <span>
        If you're here, you must have received your invitation, so please use the name on your card and the code to contribute to our giftlist.
      </span>
    </p>
  </div>
  <div
    className="main_content_thank_you_message"
  >
    <span>
      Thank you for being part of our story!
    </span>
  </div>
  <form
    className="main_content_form"
  >
    <div
      className="main_content_form_item"
    >
      <label
        htmlFor="fullName"
      >
        <span>
          Name:
        </span>
      </label>
      <input
        data-test-input-name={true}
        name="fullName"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="main_content_form_item"
    >
      <label
        htmlFor="secretCode"
      >
        <span>
          Secret code:
        </span>
      </label>
      <input
        data-test-input-code={true}
        name="secretCode"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div>
      <button
        className="main_content_button"
        data-test-submit-button={true}
        onClick={[Function]}
        type="submit"
      >
        <span>
          View payment option
        </span>
      </button>
    </div>
  </form>
</div>
`;
